### PR TITLE
feat: Add support for server-side cursors in postgresql module

### PR DESCRIPTION
**Pull request by kukushking**
### Relates
- #2230 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: awswrangler/postgresql.py [hunk 1]
```diff
@@ -73,6 +74,41 @@ def _create_table(
...
+) -> Iterator[pd.DataFrame]:
+    with con.cursor() as cursor:
+        sscursor_name: str = f"c_{uuid.uuid4().hex}"
+        cursor_args = _db_utils._convert_params(f"DECLARE {sscursor_name} CURSOR FOR {sql}", params)
+        cursor.execute(*cursor_args)
+
+        try:
+            while True:
+                cursor.execute(f"FETCH FORWARD {chunksize} FROM {sscursor_name}")
+                records = cursor.fetchall()
```
Comment (kukushking) [inline awswrangler/postgresql.py:95]: Pg8000 is not fully DB API 2.0 - compliant with `fetchmany()` fetching all results but returning only x. This PR is a workaround to allow fetching only specific amount of results using server-side cursors.

Ultimately we'd like pg8000 to add full support for `fetchmany()` or add `SSCursor` implementation similar to MySQL, hiding the implementation details, but until it's there this workaround addresses the problem. I might do a PR to pg8000 later to address this.

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: c0788a93d3f54d451e6dd4555dfa1a81c69be61d
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=9f57d34f-3ec1-4b47-9a46-fa0489365e9a%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 8984fcbe85dc3284303e85d6e573b3609bf32f85
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=81b804f7-fd4d-4750-9bf4-9c2896fe59dd%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 23a2837eb0ae794295ec5466b65f46ecb99549bf
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=f46e528d-ec20-4cda-a2f1-9ec3b828517a%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 23a2837eb0ae794295ec5466b65f46ecb99549bf
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=8b69e09b-1c54-41e0-b3be-4405a3b586c1%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by LeonLuttenberger — APPROVED** [review]
Awesome work! Left one minor comment.

File: awswrangler/postgresql.py (hunk 1 above)
Comment (kukushking) [reply awswrangler/postgresql.py:102]: Good, idea, thanks! Added

**Review by jaidisido — APPROVED** [review]

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: ed6f89dd8b9931335122eac3a9d5457537da791c
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c912a74d-f406-48fa-bd12-04986a62325e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: ed6f89dd8b9931335122eac3a9d5457537da791c
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=98e50c49-73b1-400b-bad6-b55b1e8c2992%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**